frontend/src/bundles/common/pages/budgets/budget-details/components/edit-budget-modal/edit-budget-modal.tsx

```diff
@@ -0,0 +1,138 @@
+import React, { useCallback } from 'react';
+import { Controller } from 'react-hook-form';
+
+import {
+    BaseModal,
+    Button,
+    Input,
+} from '~/bundles/common/components/components';
+import { ButtonSize } from '~/bundles/common/enums/button-size.enum';
+import { ButtonVariant } from '~/bundles/common/enums/button-variant.enum';
+import { useAppForm } from '~/bundles/common/hooks/hooks';
+
+import {
+    RenderCurrency,
+    RenderDate,
+    RenderMultiDropdown,
+    RenderRecurrence,
+} from '../components';
+import styles from './styles.module.scss';
+
+interface FormData {
+    name: string;
+    amount: number;
+    currency: string;
+    recurrence: string;
+    categories: {
+        id: string;
+    }[];
+    startDate: string;
+}
+
+interface EditBudgetModalProperties {
+    isShown: boolean;
+    onClose: () => void;
+    budget: FormData;
+}
+
+const EditBudgetModal = ({
+    isShown,
+    onClose,
+    budget,
+}: EditBudgetModalProperties): JSX.Element => {
+    const { control, errors, handleSubmit, trigger } = useAppForm({
+        defaultValues: {
+            name: budget.name,
+            amount: budget.amount,
+            currency: budget.currency,
+            recurrence: budget.recurrence,
+            categories: budget.categories,
+            startDate: budget.startDate,
+        },
+    });
+
+    const handleUpdateBudgetSubmit = useCallback((formData: FormData): void => {
+        // eslint-disable-next-line no-console
+        console.log(formData);
```

Comment:
Why we need this console.log?

Reply:
It resolved in [https://github.com/BinaryStudioAcademy/bsa-winter-2022-2023-smartspend/pull/301/files#diff-239043c4b2074e50009c2ae567f6825686d1a2f949877d17e0dad66f1354852f](url)